File: src/test/java/javabot/mock/SayOperationsTest.java
package javabot.mock;

import javabot.BaseMessagingTest;
import org.testng.annotations.Test;

@Test(groups = {"operations"})
public class SayOperationsTest extends BaseMessagingTest {
    public void testSay() {
        testMessage("~say MAGNIFICENT", "MAGNIFICENT");
    }
}

File: src/test/kotlin/javabot/operations/SayOperationsTest.kt
package javabot.operations

import jakarta.inject.Inject
import javabot.BaseTest
import javabot.MockIrcAdapter
import org.testng.Assert.assertEquals
import org.testng.annotations.Test

@Test(groups = arrayOf("operations"))
class SayOperationsTest : BaseTest() {
    @Inject private lateinit var operation: SayOperation

    fun testSay() {
        val response = operation.handleMessage(message("~say MAGNIFICENT"))
        assertEquals(response.size, 1)
        assertEquals(response[0].value, "MAGNIFICENT")
    }

    fun testSayNoOp() {
        val mockIrcAdapter = bot.get().adapter as MockIrcAdapter
        mockIrcAdapter.disableOperation("isOp")

        val response = operation.handleMessage(message("~say MAGNIFICENT"))
        assertEquals(response.size, 0)
    }
}
